openstack/compute/v2/servers/requests.go

```diff
@@ -182,6 +182,7 @@ type CreateOpts struct {
 
 // ToServerCreateMap assembles a request body based on the contents of a CreateOpts.
 func (opts CreateOpts) ToServerCreateMap() (map[string]interface{}, error) {
+	var serverCreateMapError error
```

Comment:
Out of the three, I think #1 is the best way to go. It resumes the "return early" format and enables booting from volume without any modifications. The error returned by OpenStack should be sufficient for the dev/user to figure out what's going on and change their gophercloud-based app accordingly.

I like it better than the change I made - I prefer sticking to the return early form.

#2 introduces a fix unique to gophercloud.

#3 feels like it's literally pushing the problem away until later.

That's just my $0.02 - I'm no dev.